### ISSUE TITLE
Open popup ?

### ISSUE THREAD

**Issue by olup** [bjerkio/oidc-react#235]
Hello ! Congrats for this start on a React port of oidc-client-js, it's a pretty good job already.

Any way to leverage the impressive set of options of oidc-client-js ? Like opening a popup ?

And also : are you open to PR ?

Cheers, thank you for this lib !

**Comment by cobraz** [comment]
Hello, thanks for submitting an issue!

Contributions are very welcome. This project has been fortunate enough to get many contributions from developers like yourself! In fact, the whole reason as to why we have a stabile release now, is the community.

You can leverage, I think, all the options from `oidc-client-js`; popup should work!

**Comment by cobraz** [comment]
I'm closing this issue now. If you have any issues or questions, let us know 🎉

**Comment by Ex4l** [comment]
> Hello ! Congrats for this start on a React port of oidc-client-js, it's a pretty good job already.
> 
> Any way to leverage the impressive set of options of oidc-client-js ? Like opening a popup ?
> 
> And also : are you open to PR ?
> 
> Cheers, thank you for this lib !

Hello @olup, did you managed to get a working app using popup ? 

I'm trying to understand how this would be implemented with oidc-react.

Maybe someone have a working example somewhere?

**Comment by olup** [comment]
If I remember correctly you can set any parameters from oidc-client-js from this lib.

**Comment by cobraz** [comment]
Hello!

I think we probably should add more documentation about this. If someone wants to contribute, I would very much appreciate that! 🙏

**Comment by Ex4l** [comment]
> Hello!
> 
> I think we probably should add more documentation about this. If someone wants to contribute, I would very much appreciate that! 🙏

I'm trying to make it working in my app currently, i'm not sure if i'm doing it correctly but the login loop is working so it's a progress (i'm having one more difficulty as my app is developed using Vercel so i can't simply declare a userManager because it's trying to use it server side 🙈 ).

What i did is using this oicdConfig item :

```typescript
  const oidcConfig: AuthProviderProps = {
    onSignIn: () => {
      router.push(localStorageRedirectUri as string).then(() => {
        sessionStorage.removeItem("oidc.origin.route");
      });
    },
    autoSignIn: false,
  };
```

The autoSignIn value is set to false otherwise i got a signinRedirect (that's where i think i'm not doing it right, is it possible to tell the config that we are using a signinPopup instead of a signinRedirect ?).

Then (for now), when i click on a button, i'm calling this method : 

```typescript
  const handleLoginClick = () => {
    const userManager = new UserManager({
      authority: process.env.AUTHORITY_URL,
      client_id: process.env.CLIENT_ID,
      popup_redirect_uri: redirectUri,
      response_type: "code",
      scope: "openid profile",
    });

    userManager.signinPopup();
  };
```
If it's the way to go i'll probably just get rid of the button and make a homemade hook just to create the userManager and calling signinPopup, if it's not and you have a simpler way to do it i'm all ears.

**Comment by cobraz** [comment]
The hook seems like something we can add instead. I think you can also utilize our Context instead of defining your own UserManager. Let me check.

**Comment by Ex4l** [comment]
Hello, i hope you didn't go to far with using your own context.

I saw that using my own UserManager was not interracting with oidc-react so i decided to do everything "by hand" instead of using the lib, creating a Login component made specially to work with Vercel (you can't even use hooks if they use localeStorage or SessionStorage with that one (joys of SSR)).

Let me know if you need some help about the lib, i'll check if i can help you.

**Comment by cobraz** [comment]
That would be amazing! 🎉

**Comment by buckstephenh** [comment]
Yes in oidc-client the userManager.signinPopup() is a different method v signinRedirect().  It certainly would be easier on this library if the signin method name was identical and the method was passed in as a prop but perhaps there is another way.  It is desirable to retain the SPA context and not do a full page redirect.

**Comment by cobraz** [comment]
This is fixed in #479. Thanks to everyone who contributed!
